packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -235,10 +234,11 @@ function BlockPatternsList(
 						showTooltip={ showTitlesAsTooltip }
 						category={ category }
 					/>
-				) : (
-					<BlockPatternPlaceholder key={ pattern.name } />
 				);
 			} ) }
+			{ shownPatterns.length < blockPatterns.length && (
+				<BlockPatternPlaceholder />
+			) }
```

Comment:
This PR removes the flickering when selecting blocks completely for me. 🥇 

I just need help in understanding the logic changes. 

In trunk, a `BlockPatternPlaceholder` is rendered in the `map()` callback for each block pattern not in `shownPatterns`.

Here, it a single `BlockPatternPlaceholder` if `shownPatterns` length is less than blockPatterns length.

Is that right?

Reply:
The block pattern renders about `20` patterns per page, and the shownPatterns are  0, 5, 10, 15 and 20 (async)
and the `BlockPatternPlaceholder` is rendered for every pattern that is not shown (20 times initially, then 15 etc..)

When it was first introduced [here](https://github.com/WordPress/gutenberg/pull/22789/files#diff-e24949b00cac037aacf5a314d639f0482c76a108cea05ebb9ad05b45345cf971), its not given any height. I assume placeholder introduced to occupy space for the pattern when it is not rendered yet. 

But in a [later change](https://github.com/WordPress/gutenberg/pull/28520/files#diff-af69062f440fdbe79fc32030507bfeb75ab712b8c42d680909e1ca1d77f7c7d4R15), the placeholder is set with 100% height.

In the current state, it is unnecessary to render 20 empty blocks with 100% height, instead rendering one single placeholder block is enough to show empty space in the initial view and until there are patterns to render. (That's what my understanding of the placeholder pattern)

It is shown until complete list of patterns are rendered and removed after.

Reply:
I have created #66114 and I think it improves the rendering much better.